### PR TITLE
model: add mixedbread-ai mxbai-rerank-base-v2 and mxbai-rerank-large-v2

### LINKED ISSUE #2360: Integrate mixedbread rerank `v2`
https://huggingface.co/collections/mixedbread-ai/reranking-series-v2-67d230aa8f3def6bcba6e887

### PR DESCRIPTION
**Pull request by sarendis56**
## model: add mixedbread-ai mxbai-rerank-v2 rerankers

Adds the [v2 reranking series](https://huggingface.co/collections/mixedbread-ai/reranking-series-v2):

- `mixedbread-ai/mxbai-rerank-base-v2` (ProRank-0.5B, built on Qwen2.5-0.5B)
- `mixedbread-ai/mxbai-rerank-large-v2` (built on Qwen2.5-1.5B)

**Implementation notes**

- `activation_fn=torch.nn.Identity()` is passed via `loader_kwargs` so the raw relevance logit is used as the score — see the ablation below for explanation.

- Extra `mxbai-rerank-v2` (`sentence-transformers>=5.4.0`, which the `LogitScore` head needs and is above mteb's core floor). It's wired into the `[tool.uv]` conflicts group, attached via `extra_requirements_groups`.

- v1 `base`/`large` entries now point to their v2 successors via `superseded_by`.

- `embed_dim` — for cross-encoders there's no embedding output, so the value to use is the transformer's hidden_size, which is the established convention for the analogous generative rerankers already in the repo (the Qwen3-Reranker entries set embed_dim to 1024/2560/4096 = their hidden sizes). I read hidden_size from each config.json:

  - mxbai-rerank-base-v2 → 896

  - mxbai-rerank-large-v2 → 1536

  - Paper related section:

    > ProRank computes a finegrained relevance score by extracting and comparing logit values corresponding to binary tokens “0” and “1” from the last token’s logits...

### v2 vs v1 (independent reproduction)

Reranking MAP, bf16. v2 beats v1 across the board:

| Task (MAP)                | large-v1 | base-v2 | large-v2   |
| ------------------------- | -------- | ------- | ---------- |
| AskUbuntuDupQuestions     | 0.6140   | 0.6049  | **0.6417** |
| SciDocsRR                 | 0.7704   | 0.7788  | **0.7783** |
| StackOverflowDupQuestions | 0.4785   | 0.5218  | **0.5341** |
| **Average**               | 0.6210   | 0.6352  | **0.6514** |

### Why identity activation (ablation)

The default `CrossEncoder` activation is sigmoid. It's monotonic, so in full precision it wouldn't change any ranking — but when the model runs in **bf16**, where `sigmoid(x)` saturates to exactly `1.0` for `x ≳ 6.2`. These rerankers emit large logits, so a large fraction of candidates collapse to a single tied score (45% on AskUbuntu, measured). mteb scores reranking via the retrieval metric, which breaks ties by document id (not relevance), so the tied-at-1.0 documents get scattered and the metric collapses.

large-v2, identical pipeline, only the activation changed:

| Task (MAP)                | identity (raw logit) | sigmoid (default) | rel. drop  |
| ------------------------- | -------------------- | ----------------- | ---------- |
| AskUbuntuDupQuestions     | 0.6417               | 0.3678            | −42.7%     |
| SciDocsRR                 | 0.7783               | 0.7613            | −2.2%      |
| StackOverflowDupQuestions | 0.5341               | 0.0994            | −81.4%     |
| **Average**               | **0.6514**           | **0.4095**        | **−37.1%** |

With sigmoid the model scores *below v1* and is near-random on StackOverflow. Identity also reproduces the model card's example scores exactly.

### Benchmark Results

Reproduced the paper's protocol — rerank BM25 top-100, **nDCG@10** with mteb's `mteb/baseline-bm25s` as the first stage:

| BEIR dataset | corpus | base-v2 (repro) | base-v2 (paper) | Δ      | large-v2 (repro) | large-v2 (paper) | Δ      |
| ------------ | ------ | --------------- | --------------- | ------ | ---------------- | ---------------- | ------ |
| TRECCOVID    | 171K   | 0.7929          | 0.7906          | +0.002 | 0.7944           | 0.8001           | −0.006 |
| NFCorpus     | 3.6K   | 0.3565          | 0.3710          | −0.015 | 0.3457           | 0.3706           | −0.025 |
| FiQA2018     | 58K    | 0.4159          | 0.4109          | +0.005 | 0.4461           | 0.4441           | +0.002 |
| ArguAna      | 8.7K   | 0.6359          | 0.5782          | +0.058 | 0.7404           | 0.6797           | +0.061 |
| Touche2020   | 382K   | 0.3283          | 0.3458          | −0.018 | 0.3505           | 0.3206           | +0.030 |
| SciFact      | 5K     | 0.7805          | 0.7715          | +0.009 | 0.7931           | 0.8087           | −0.016 |

Close https://github.com/embeddings-benchmark/mteb/issues/2360

### Validation

- [x] I have filled out the ModelMeta object to the extent possible
- [x] I have ensured that my model can be loaded using
  - [x] `mteb.get_model(model_name, revision)` and
  - [x] `mteb.get_model_meta(model_name, revision)`
- [x] I have tested the implementation works on a representative set of tasks.
- [x] The model is public, i.e., is available either as an API or the weights are publicly available to download
- [x] I reproduced results from the original paper (if applicable) on at least one benchmark, and I am including the results in the PR description.

### CODE REVIEW

File: mteb/models/model_implementations/mixedbread_ai_models.py [hunk 1]
```diff
@@ -244,6 +246,201 @@
...
+    framework=["PyTorch", "Sentence Transformers", "Transformers", "safetensors"],
+    reference="https://huggingface.co/mixedbread-ai/mxbai-rerank-large-v2",
+    similarity_fn_name=None,
+    use_instructions=False,
+    training_datasets=mixedbread_training_data,
+    adapted_from="Qwen/Qwen2.5-1.5B",
+    superseded_by=None,
+    modalities=["text"],
+    model_type=["cross-encoder"],
+    extra_requirements_groups=("mxbai-rerank-v2",),
```
Comment (Samoed) [inline mteb/models/model_implementations/mixedbread_ai_models.py]: I think you can use `multimodal-sbert` here. On what Transformers version did you run your evaluations?
```suggestion
    extra_requirements_groups=("multimodal-sbert",),
```

**Review by Samoed — APPROVED** [review]

File: mteb/models/model_implementations/mixedbread_ai_models.py (hunk 1 above)
Comment (sarendis56) [reply mteb/models/model_implementations/mixedbread_ai_models.py]: I validated on transformers 5.10.2 / sentence‑transformers 5.5.1. The multimodal-sbert would work. Let me run a quick trial

Comment (sarendis56) [reply mteb/models/model_implementations/mixedbread_ai_models.py]: Good. It works as well. Now this PR would not change the project toml or uv.lock!
